compiler/internal/codegen/codegen_main.go

```diff
@@ -754,9 +754,14 @@ func (b *Builder) writeDecoder(f *File) {
 	f.Line()
 }
 
-func (b *Builder) typeName(param *est.Param, skipPtr bool) Code {
-	decl := param.Decl
-	typName := Qual(decl.Loc.PkgPath, decl.Name)
+func (b *Builder) typeName(param *est.Param, skipPtr bool) *Statement {
+	// Reconstruct a named type to make the schemaTypeToGoType function slightly easier to recursive over
```

Comment:
Not sure I understand this part

Reply:
The next PR will update this as discussed last night